packages/kbn-std/src/is_internal_url.test.ts

```diff
@@ -86,5 +103,34 @@ describe('isInternalURL', () => {
       const hrefWithThreeSlashes = `///app/kibana`;
       expect(isInternalURL(hrefWithThreeSlashes)).toBe(false);
     });
+
+    it('should return `false` if absolute URL contains tabs or new lines', () => {
+      for (const char of ['\t', '\n', '\r', '\t\n\r']) {
+        for (const url of [
+          `/${char}/app/kibana`,
+          `/${char}//app/kibana`,
+          `//${char}/app/kibana`,
+          `/${char}/example.com`,
+          `/${char}//example.com`,
+          `//${char}/example.com`,
+          `htt${char}ps://example.com`,
+          `/${char}/example.org`,
+          `/${char}//example.org`,
+          `//${char}/example.org`,
+          `htt${char}ps://example.org`,
+          `/${char}/example.org:5601`,
+          `/${char}//example.org:5601`,
+          `//${char}/example.org:5601`,
+          `htt${char}ps://example.org:5601`,
+          `java${char}script:alert(1)`,
+          `htt${char}p://example.org:5601`,
+          `https:/${char}/example.com`,
+          `file${char}://example.com`,
+          `https://example${char}.com/path`,
+        ]) {
```

Comment:
Same here. These would all return false, even without the special character. Should we have at least one test case that would otherwise return true, if not for the tab/newline?

I tried this with using '/app/kibana#/discover/New-Saved-Search' (from the true test case) and adding the special char in various places, but it still returned true. Am I misinterpreting the test?

Reply:
> These would all return false, even without the special character. 

Without the special character, they will behave as expected with both the current and previous code and return `false`, and we have existing tests for this behavior. The issue I'm trying to cover here is that with the previous code, the presence of the special character made this function mistakenly return `true` instead of `false`. Try reverting the changes in `is_internal_url.ts` but keep the tests.

Does that make sense?

> Should we have at least one test case that would otherwise return true, if not for the tab/newline?

Yeah, I think it's a good idea. What about having a similar loop with a subset of the URLs tested here, but with a white space instead of `\t\r\n` that should cause the function to return `true`?

Reply:
Okay, I've made an attempt to make the intention a bit clearer in unit tests in https://github.com/elastic/kibana/pull/183521/commits/96772c33b7fe3a85aef89e33a1b23d534e59e96b. @jeramysoucy let me know if it helps.

Reply:
Yes, this is much more clear to me now. Thanks, Oleg!